types/gsi/index.d.ts

```diff
@@ -0,0 +1,271 @@
+     *
+     * ```
+     * <script>
+     * function onSignout() {
+     *   google.accounts.id.disableAutoSelect()
+     * }
+     * </script>
+     * ```
+     */
+    function disableAutoSelect(): void;
+
+    /**
+     * This method is a simple wrapper for the store() method of the browser's native credential manager API. Therefore, it can only be used to store a password credential.
+     *
+     * The following code example implements the google.accounts.id.storeCredential method with an onSignIn() function:
+     *
+     * ```
+     * <script>
+     * function onSignIn() {
+     *   let cred = {id: '...', password: '...'}
+     *   google.accounts.id.storeCredential(cred)
+     * }
+     * </script>
+     * ```
+     *
+     * @param credential Credential object.
+     * @param callback Optional callback to run code after the credentials are stored.
+     */
+    function storeCredential(credential: GsiCredential, callback?: () => void): void;
+
+    /**
+     * You can cancel the One Tap flow if you remove the prompt from the relying party DOM. The cancel operation is ignored if a credential is already selected.
+     *
+     * The following code example implements the google.accounts.id.cancel() method with an onNextButtonClicked() function:
+     *
+     * ```
+     * <script>
+     * function onNextButtonClicked() {
+     *   google.accounts.id.cancel()
+     *   showPasswordPage()
+     * }
+     * </script>
+     * ```
+     */
+    function cancel(): void;
+
+    /**
+     * The google.accounts.id.revoke method revokes the OAuth grant used to share the ID token for the specified user.
+     *
+     * The following code sample shows how use the revoke method with an Id.
+     *
+     * ```
+     * google.accounts.id.revoke('1618033988749895', done => {
+     *   console.log(done.error)
+     * })
+     * ```
+     * @param hint The email address or unique ID of the user's Google Account. The ID is the sub property of the credential payload.
+     * @param callback Optional RevocationResponse handler.
+     */
+    function revoke(hint: string, callback?: (done: RevocationResponse) => void): void;
+}
+
+interface Window {
+    /**
+     * You can register an onGoogleLibraryLoad callback. It's notified after the Sign In With Google JavaScript library is loaded:
+     *
+     * ```
+     * window.onGoogleLibraryLoad = () => {
+     *   // ...
+     * }
+     * ```
+     *
+     * This callback is just a shortcut for the window.onload callback. There are no differences in behavior.
+     *
+     * The following code example implements an onGoogleLibraryLoad callback:
+     *
+     * ```
+     * <script>
+     * window.onGoogleLibraryLoad = () => {
+     *   google.accounts.id.initialize({
+     *     // ...
+     *   })
+     *   google.accounts.id.prompt()
+     * }
+     * </script>
+     * ```
+     */
+    onGoogleLibraryLoad: () => void;
```

Comment:
this one can be undefined, IMO declaration should allow to check (real life):
```ts
if (!window.onGoogleLibraryLoad) {...
```

Reply:
good point. updated!

Reply:
good point. updated!